CDP4Reporting/DataSource/ReportingDataSourceParameter.cs

```diff
@@ -0,0 +1,124 @@
+        /// </summary>
+        protected DomainOfExpertise Owner => this.ParameterBase?.Owner;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReportingDataSourceParameter{T}"/> class.
+        /// </summary>
+        protected ReportingDataSourceParameter()
+        {
+            this.ShortName = GetParameterAttribute(this.GetType())?.ShortName;
+        }
+
+        /// <summary>
+        /// Initializes a reported parameter column based on the corresponding
+        /// <see cref="CDP4Common.EngineeringModelData.ParameterBase"/> within the associated
+        /// <see cref="ReportingDataSourceNode{T}"/>.
+        /// </summary>
+        /// <param name="node">
+        /// The associated <see cref="ReportingDataSourceNode{T}"/>.
+        /// </param>
+        internal override void Initialize(ReportingDataSourceNode<T> node)
+        {
+            this.Node = node;
+
+            this.ParameterBase = this.ParameterBase ??
+                                 this.Node.NestedElement.NestedParameter.SingleOrDefault(
+                                     x => x.AssociatedParameter.ParameterType.ShortName == this.ShortName)?.AssociatedParameter;
+
+            this.ParameterBase = this.ParameterBase ??
+                                 this.Node.ElementUsage?.ParameterOverride.SingleOrDefault(
+                                     x => x.Parameter.ParameterType.ShortName == this.ShortName);
+
+            this.ParameterBase = this.ParameterBase ??
+                                 this.Node.ElementDefinition.Parameter.SingleOrDefault(
+                                     x => x.ParameterType.ShortName == this.ShortName);
+
+            this.Value = this.ParameterBase?.ValueSets.First().ActualValue.First();
+        }
+
+        /// <summary>
+        /// Gets the parameter of type <see cref="TP"/> on the same level in the
+        /// hierarhical tree upon which the data source is based.
+        /// </summary>
+        /// <typeparam name="TP">
+        /// The desired parameter type.
+        /// </typeparam>
+        /// <returns>
+        /// The <see cref="ReportingDataSourceParameter{T}"/> of type <see cref="TP"/>.
+        /// </returns>
+        public TP GetSibling<TP>() where TP : ReportingDataSourceParameter<T>
+        {
+            return this.Node.GetColumn<TP>();
+        }
+
+        /// <summary>
+        /// Gets the parameters of type <see cref="TP"/> on the children levels in the
+        /// hierarhical tree upon which the data source is based.
+        /// </summary>
+        /// <typeparam name="TP">
+        /// The desired parameter type.
+        /// </typeparam>
+        /// <returns>
+        /// A list of <see cref="ReportingDataSourceParameter{T}"/>s of type <see cref="TP"/>.
+        /// </returns>
+        public IEnumerable<TP> GetChildren<TP>() where TP : ReportingDataSourceParameter<T>
```

Comment:
The intent for the `GetSibling`/`GetChildren` methods was to have the option to compute some composite parameter values in the data source as well, not only in the report itself. Given the latest discussions, it seems like this is no longer desired.

I suppose these methods could now be removed entirely, but it probably warrants a separate discussion.

Reply:
Althought the original 10-25 "tree structure" is not completely present in the forest of ReportingDataSourceNodes, I still think traversing the Children could be interesting. Functionality needed by GetSibling can be done in the DataSource C# editor. I'll remove GetSibling and let GetChildren stay.